packages/api/src/external/fhir/document/index.ts

```diff
@@ -189,48 +202,70 @@ export function idToFHIR(id: DocumentIdentifier): Identifier {
+export function convertToFHIRResource(
+  resource: Contained,
+  patientId: string
+): Resource[] | undefined {
   if (resource.resourceType === "Patient" && resource.id && !resource.id.includes(patientId)) {
     capture.message(`Found a Patient resource with a different ID`, {
       extra: { context: `toFHIR.convertToFHIRResource`, resource, patientId, level: "warning" },
     });
-    return {
-      resourceType: "Patient",
-      id: resource.id,
-      address: convertCWAdressToFHIR(resource.address),
-      gender: convertCWGenderToFHIR(resource.gender?.coding),
-      identifier: convertCWIdentifierToFHIR(resource.identifier),
-      name: resource.name ? convertCWNameToHumanName(resource.name) : undefined,
-    };
+    return [
+      {
+        resourceType: "Patient",
+        id: resource.id,
+        address: convertCWAdressToFHIR(resource.address),
+        gender: convertCWGenderToFHIR(resource.gender?.coding),
+        identifier: convertCWIdentifierToFHIR(resource.identifier),
+        name: resource.name ? convertCWNameToHumanName(resource.name) : undefined,
+      },
+    ];
   } else if (resource.resourceType === "Organization" && resource.name) {
-    return {
-      resourceType: "Organization",
-      id: resource.id ?? undefined,
-      identifier: convertCWIdentifierToFHIR(resource.identifier),
-      name: convertCWNameToString(resource.name),
-      address: convertCWAdressToFHIR(resource.address),
-    };
+    return [
+      {
+        resourceType: "Organization",
+        id: resource.id ?? undefined,
+        identifier: convertCWIdentifierToFHIR(resource.identifier),
+        name: convertCWNameToString(resource.name),
+        address: convertCWAdressToFHIR(resource.address),
+      },
+    ];
   } else if (resource.resourceType === "Practitioner" && resource.name) {
-    return {
+    const practitioner: Resource = {
       resourceType: "Practitioner",
       id: resource.id ?? undefined,
       identifier: convertCWIdentifierToFHIR(resource.identifier),
       name: convertCWNameToHumanName(resource.name),
       gender: convertCWGenderToFHIR(resource.gender?.coding),
     };
-  } else {
-    capture.message(
-      `New Resource type on toFHIR conversion - might need to handle in CW doc ref mapping`,
-      {
-        extra: {
-          context: `toFHIR.convertToFHIRResource`,
-          resourceType: resource.resourceType,
-          resource,
-          patientId,
-        },
-      }
-    );
+    const role: Resource | undefined =
+      resource.organization?.reference && resource.id
+        ? {
+            resourceType: "PractitionerRole",
+            organization: {
+              type: "Organization",
+              reference: resource.organization.reference,
+            },
+            practitioner: {
+              type: "Practitioner",
+              reference: `#${resource.id}`,
+            },
+          }
+        : undefined;
+    return [practitioner, ...(role ? [role] : [])];
```

Comment:
why is this being included?

https://www.hl7.org/fhir/R4/practitionerrole.html "A specific set of Roles/Locations/specialties/services that a practitioner may perform at an organization for a period of time."

this doesn't seem to add any new information?

Reply:
This info comes from CW and is mapped as part of the scope of the ticket: https://github.com/metriport/metriport-internal/issues/1033#issuecomment-1711056833

I'm happy to review it, but it does seem valuable, especially when considering we might have diff Practitioners coming in `contained`, each associated to a diff Org.

Thoughts?

Reply:
I see - from the perspective of not losing info/context from the doc ref it makes sense.

Note that it seems the possibility of multiple different orgs having different practitioners in `contained` is low, as AFAIK we've only seen a single org per contained. However, acknowledged that there is no guarantee that this will always be the case.